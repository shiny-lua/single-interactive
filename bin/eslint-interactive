#!/usr/bin/env node -r source-map-support/register

require('../dist');
