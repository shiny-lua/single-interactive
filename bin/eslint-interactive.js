#!/usr/bin/env -S node --enable-source-maps --unhandled-rejections=strict --experimental-import-meta-resolve

import { run } from '../dist/index.js';

run({
  argv: process.argv,
}).catch((error) => {
  console.error(error);
  process.exit(1);
});
