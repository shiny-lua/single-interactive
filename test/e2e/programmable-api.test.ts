import { dirname, join } from 'path';
import { fileURLToPath } from 'url';
import { Core, takeRuleStatistics } from '@mizdra/eslint-interactive';
import stripAnsi from 'strip-ansi';

test('Programmable API', async () => {
  const core = new Core({
    patterns: ['fixtures'],
    cwd: join(dirname(fileURLToPath(import.meta.url)), '..', '..'),
  });
  const results = await core.lint();

  expect(stripAnsi(core.formatResultSummary(results))).toMatchSnapshot();
  const statistics = takeRuleStatistics(results);
  expect(statistics).toMatchSnapshot();

  const sortedStatistics = statistics
    // Exclude non-fixable statistic
    .filter((statistic) => statistic.isFixableCount > 0)
    // Sort by descending order of fixable count
    .sort((a, b) => b.isFixableCount - a.isFixableCount);

  const ruleIds = sortedStatistics.map((statistic) => statistic.ruleId);

  const top3RuleIds = ruleIds.slice(0, 3);

  expect(top3RuleIds).toMatchInlineSnapshot(`
    Array [
      "semi",
      "prefer-const",
      "import/order",
    ]
  `);
});
