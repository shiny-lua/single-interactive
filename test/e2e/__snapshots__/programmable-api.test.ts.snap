// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Programmable API 1`] = `
"- 9 files (0 file passed, 9 files failed) checked.
- 48 problems (47 errors, 1 warning) found.
┌─────────────────────────────┬───────┬─────────┬────────────┬─────────────────┐
│ Rule                        │ Error │ Warning │ is fixable │ has suggestions │
├─────────────────────────────┼───────┼─────────┼────────────┼─────────────────┤
│ ban-exponentiation-operator │ 9     │ 0       │ 0          │ 0               │
├─────────────────────────────┼───────┼─────────┼────────────┼─────────────────┤
│ semi                        │ 14    │ 1       │ 15         │ 0               │
├─────────────────────────────┼───────┼─────────┼────────────┼─────────────────┤
│ import/order                │ 3     │ 0       │ 3          │ 0               │
├─────────────────────────────┼───────┼─────────┼────────────┼─────────────────┤
│ prefer-const                │ 4     │ 0       │ 4          │ 0               │
├─────────────────────────────┼───────┼─────────┼────────────┼─────────────────┤
│ no-unused-vars              │ 7     │ 0       │ 0          │ 0               │
├─────────────────────────────┼───────┼─────────┼────────────┼─────────────────┤
│ no-useless-escape           │ 5     │ 0       │ 0          │ 5               │
├─────────────────────────────┼───────┼─────────┼────────────┼─────────────────┤
│ no-unsafe-negation          │ 5     │ 0       │ 0          │ 5               │
└─────────────────────────────┴───────┴─────────┴────────────┴─────────────────┘"
`;

exports[`Programmable API 2`] = `
Array [
  Object {
    "errorCount": 9,
    "hasSuggestionsCount": 0,
    "hasSuggestionsErrorCount": 0,
    "hasSuggestionsWarningCount": 0,
    "isFixableCount": 0,
    "isFixableErrorCount": 0,
    "isFixableWarningCount": 0,
    "ruleId": "ban-exponentiation-operator",
    "warningCount": 0,
  },
  Object {
    "errorCount": 14,
    "hasSuggestionsCount": 0,
    "hasSuggestionsErrorCount": 0,
    "hasSuggestionsWarningCount": 0,
    "isFixableCount": 15,
    "isFixableErrorCount": 14,
    "isFixableWarningCount": 1,
    "ruleId": "semi",
    "warningCount": 1,
  },
  Object {
    "errorCount": 3,
    "hasSuggestionsCount": 0,
    "hasSuggestionsErrorCount": 0,
    "hasSuggestionsWarningCount": 0,
    "isFixableCount": 3,
    "isFixableErrorCount": 3,
    "isFixableWarningCount": 0,
    "ruleId": "import/order",
    "warningCount": 0,
  },
  Object {
    "errorCount": 4,
    "hasSuggestionsCount": 0,
    "hasSuggestionsErrorCount": 0,
    "hasSuggestionsWarningCount": 0,
    "isFixableCount": 4,
    "isFixableErrorCount": 4,
    "isFixableWarningCount": 0,
    "ruleId": "prefer-const",
    "warningCount": 0,
  },
  Object {
    "errorCount": 7,
    "hasSuggestionsCount": 0,
    "hasSuggestionsErrorCount": 0,
    "hasSuggestionsWarningCount": 0,
    "isFixableCount": 0,
    "isFixableErrorCount": 0,
    "isFixableWarningCount": 0,
    "ruleId": "no-unused-vars",
    "warningCount": 0,
  },
  Object {
    "errorCount": 5,
    "hasSuggestionsCount": 5,
    "hasSuggestionsErrorCount": 5,
    "hasSuggestionsWarningCount": 0,
    "isFixableCount": 0,
    "isFixableErrorCount": 0,
    "isFixableWarningCount": 0,
    "ruleId": "no-useless-escape",
    "warningCount": 0,
  },
  Object {
    "errorCount": 5,
    "hasSuggestionsCount": 5,
    "hasSuggestionsErrorCount": 5,
    "hasSuggestionsWarningCount": 0,
    "isFixableCount": 0,
    "isFixableErrorCount": 0,
    "isFixableWarningCount": 0,
    "ruleId": "no-unsafe-negation",
    "warningCount": 0,
  },
]
`;
